docs/playbook/ingesting-data/013-creating-a-lambda-function-step-by-step.md

```diff
@@ -0,0 +1,237 @@
+---
+id: creating-a-lambda-function-step-by-step
+title: How to create a Lambda function for data ingestion on the Data Analytics Platform
+description: "Ingesting data using AWS Lambda [step-by-step]"
+layout: playbook_js
+tags: [playbook]
+---
+
+# How to create a Lambda function for data ingestion on the Data Analytics Platform
+
+### Check the feasibility of a Lambda function approach
+
+Before starting, it is advisable to discuss with the DAP team that a Lambda function is the most appropriate approach to take for your data ingestion task.
+
+Here are some factors to consider before proceeding:
+
+* _Execution Time_
+
+    Lambda functions have execution time limits (currently 15 minutes per invocation). Ensure that your workload can execute within this time limit. If not, have a discussion with the DAP team to identify a more suitable approach.
+
+
+* _Supported Runtimes and dependencies_
+
+    Lambda supports multiple programming languages including Python. Certain Python packages may need to be imported. This can be achieved through Lambda layers.
+
+
+* _Monitoring_
+    
+    Lambda integrates with [AWS CloudWatch](https://docs.aws.amazon.com/lambda/latest/dg/monitoring-cloudwatchlogs.html) for logging and monitoring; ensure that the permissions set includes the ability to write logs for debugging later on.
+
+### Outline what you need your Lambda function to do
+
+On the Data Analytics Platform, we typically use [AWS Lambda](https://docs.aws.amazon.com/lambda/latest/dg/welcome.html) to perform simple ingestion and data processing tasks which do not require the resource-intensive infrastructure of an [AWS Glue job](https://docs.aws.amazon.com/glue/latest/dg/what-is-glue.html), for example:
+
+* Making an _API call_ and writing the response to the landing zone, including a crawler execution
+* _Reading_ in data from a _Google Sheet_
+* _Moving data_ from one zone to another zone, with a simple refinement
+
```

Comment:
Event-Driven or Real-Time Processing are classic use cases. For example, once data arrive in S3, we may trigger the Lambda to do something immediately; when we add someone to a Google group, then trigger the Lambda function to grant this permission in near real time.